### PR TITLE
Add noop argument to bypass agent logic

### PR DESCRIPTION
**Pull request by boy-uber**

### CODE REVIEW

File: src/test/java/com/uber/profiling/ArgumentsTest.java [hunk 1]
```diff
@@ -57,6 +59,14 @@ public void emptyArgumentValue() {
         Arguments.parseArgs("reporter=,durationProfiling=,metricInterval=,appIdRegex=,");
     }
 
+    @Test
+    public void noop() {
+        Arguments arguments = Arguments.parseArgs("durationProfiling=a.bc.foo,noop=true,durationProfiling=ab.c.d.test");
+        Assert.assertEquals(2, arguments.getRawArgValues().size());
+        Assert.assertTrue(arguments.isNoop());
+        Assert.assertEquals(2, arguments.getDurationProfiling().size());
+    }
```
Comment (CodingCat) [inline src/test/java/com/uber/profiling/ArgumentsTest.java]: do we need to have a test to verify when noop is set, no profiling would be executed?

**Comment by felixcheung** [comment]
I don’t completely get why we need this. Is there information on this?

File: src/test/java/com/uber/profiling/ArgumentsTest.java (hunk 1 above)
Comment (boy-uber) [reply src/test/java/com/uber/profiling/ArgumentsTest.java]: Good suggestion! Will add a test.

**Comment by boy-uber** [comment]
@felixcheung Because we add ConfigProvider and allow people to change config from a server side HTTP config file. Adding noop will allow people to "turn off" the agent by changing the server side config.

**Comment by CodingCat** [comment]
> @felixcheung Because we add ConfigProvider and allow people to change config from a server side HTTP config file. Adding noop will allow people to "turn off" the agent by changing the server side config.

following this, how people would turn off a specific profiler?

**Pull request merged**